pkg/server/helpers.go

```diff
@@ -190,18 +190,29 @@ func criContainerStateToString(state runtime.ContainerState) string {
 }
 
 // getRepoDigestAngTag returns image repoDigest and repoTag of the named image reference.
-func getRepoDigestAndTag(namedRef reference.Named, digest imagedigest.Digest, schema1 bool) (string, string) {
-	var repoTag, repoDigest string
+func getRepoDigestAndTag(namedRef reference.Named, repoRef reference.Named, digest imagedigest.Digest, schema1 bool) ([]string, []string) {
```

Comment:
Are you sure docker keeps both repo tags and both repo digests for original ref and mirror ref? I remember that is not the case when you showed it to me.

Reply:
Here is what I get from `docker pull google_containers/node-problem-detector:v0.4.1` with registry mirror `https://gcr.io/`
```console
[
    {
        "Id": "sha256:cf5601c20da55c9e76cd4cb85d4e5c5eb888c7babf73be6903e38b91709d7ef7",
        "RepoTags": [
            "google_containers/node-problem-detector:v0.4.1"
        ],
        "RepoDigests": [
            "google_containers/node-problem-detector@sha256:f95cab985c26b2f46e9bd43283e0bfa88860c14e0fb0649266babe8b65e9eb2b"
        ],
        "Parent": "",
        "Comment": "",
        "Created": "2017-06-21T23:02:03.851211741Z",
        "Container": "f50da647cf59c808c045a004a3cbd95bbb815c22b37b4a178f12260e4cf1bc72",
        "ContainerConfig": {
            "Hostname": "c10bc3f0ac50",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "DISTTAG=f25docker",
                "FGC=f25",
                "FBR=f25"
            ],
            "Cmd": [
                "/bin/sh",
                "-c",
                "#(nop) ",
                "ENTRYPOINT [\"/node-problem-detector\" \"--system-log-monitors=/config/kernel-monitor.json\"]"
            ],
            "Image": "sha256:86876eef969b99d8c348c031c5c1ae715c3700e5d5fe81392b43183c7344f433",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": [
                "/node-problem-detector",
                "--system-log-monitors=/config/kernel-monitor.json"
            ],
            "OnBuild": [],
            "Labels": {}
        },
        "DockerVersion": "1.12.6",
        "Author": "Random Liu \u003clantaol@google.com\u003e",
        "Config": {
            "Hostname": "c10bc3f0ac50",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "DISTTAG=f25docker",
                "FGC=f25",
                "FBR=f25"
            ],
            "Cmd": null,
            "Image": "sha256:86876eef969b99d8c348c031c5c1ae715c3700e5d5fe81392b43183c7344f433",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": [
                "/node-problem-detector",
                "--system-log-monitors=/config/kernel-monitor.json"
            ],
            "OnBuild": [],
            "Labels": {}
        },
        "Architecture": "amd64",
        "Os": "linux",
        "Size": 286572743,
        "VirtualSize": 286572743,
        "GraphDriver": {
            "Name": "aufs",
            "Data": null
        },
        "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:bf973ec11f37bce787f28f7ccf15aa42e6c20188c5267caea4973f802edec873",
                "sha256:32c90b788a552ca2002c54bf342668a96cc558366674e00e44226a509e30d733",
                "sha256:72cad11943987e17eb5adc147b2cf1dd5bc5c259c34aecfb370868570f21b05d",
                "sha256:ff41cf9c6728f7c01c643b6003d29bcafc4d25b0dda20ea1a0031480821d6fc9"
            ]
        }
    }
]
root@workstation:~# docker verison
```

Reply:
Note that I'm using docker 1.12.3.